### PR TITLE
Feature/updateDataCenterAPI

### PR DESCRIPTION
**Pull request by Azher2Ali**
Implemented a PATCH endpoint for partial updates of Data Centers, ensuring that only provided values are modified and handling immutable properties appropriately. The endpoint returns a JSON object with the updated Data Center. Error handling includes returning a 400 error for invalid properties and a 404 error if the provided short name is not found. Authentication restricts access to DCC Admins and RDPC Admins with proper permissions, and unit testing was conducted for validation.

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java [hunk 1]
```diff
@@ -362,4 +363,18 @@ public List<DataCenterDTO> listDataCenters() {
         .map(s -> dataCenterConverter.dataCenterToDataCenterEntity(s))
         .collect(Collectors.toList());
   }
+
+  public DataCenterDTO createDataCenter(DataCenterRequestDTO dataCenterRequestDTO) {
+    val dataCenterEntity = programService.createDataCenter(dataCenterRequestDTO);
+    return dataCenterConverter.dataCenterToDataCenterEntity(dataCenterEntity);
+  }
+
+  @Transactional
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java:372]: Good idea, very nice.

**Review by joneubank — CHANGES_REQUESTED** [review]
Mostly good stuff, requesting one easy fix on request paths, plus one change to clean up the request bodies for create vs update requests.

File: src/main/java/org/icgc/argo/program_service/controller/DataCenterController.java [hunk 2]
```diff
@@ -30,15 +30,38 @@ public ResponseEntity<List<DataCenterDTO>> listDataCenters(
     val dataCenterEntities = serviceFacade.listDataCenters();
     return new ResponseEntity(dataCenterEntities, HttpStatus.OK);
   }
-  
- @GetMapping(value = "/datacenters/{datacenter_short_name}/programs")
+
+  @GetMapping(value = "/datacenters/{datacenter_short_name}/programs")
```
[the remark this answers is not in the document]
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/controller/DataCenterController.java]: Removed datacenter from API path

File: src/main/java/org/icgc/argo/program_service/model/dto/DataCenterRequestDTO.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  private String analysisSongUrl;
+
+  private String analysisScoreUrl;
+
+  private String submissionSongCode;
+
+  private String submissionSongUrl;
+
+  private String submissionScoreUrl;
+}
```
[the remark this answers is not in the document]
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/model/dto/DataCenterRequestDTO.java:57]: Validating the create request through ValidationService, therefore we can still use the same request DTO for create and update.

**Review by joneubank — APPROVED** [review]
This looks clean. Thanks for the updates!

**Pull request merged**